test/test_dynamic_shape_models.py

```diff
@@ -0,0 +1,93 @@
+import sys
+import numpy as np
+import unittest
+import torch
+import torch_xla.core.xla_model as xm
+import torch_xla.debug.metrics as met
+
+xla_dev = xm.xla_device()
+
+
+class Feedforward(torch.nn.Module):
+
+  def __init__(self, input_size, hidden_size):
+    super().__init__()
+    self.input_size = input_size
+    self.hidden_size = hidden_size
+    self.fc1 = torch.nn.Linear(self.input_size, self.hidden_size)
+    self.fc1.weight.data.fill_(0.01)
+    self.fc1.bias.data.fill_(0.01)
+    self.relu = torch.nn.ReLU()
+    self.fc2 = torch.nn.Linear(self.hidden_size, 1)
+    self.fc2.weight.data.fill_(0.01)
+    self.fc2.bias.data.fill_(0.01)
+    self.sigmoid = torch.nn.Sigmoid()
+
+  def forward(self, x):
+    hidden = self.fc1(x)
+    relu = self.relu(hidden)
+    output = self.fc2(relu)
+    output = self.sigmoid(output)
+    return output
+
+
+@unittest.skipIf(
+    xm.get_xla_supported_devices("CPU"),
+    f"The tests fail on CPU. See https://github.com/pytorch/xla/issues/4298 for more detail."
+)
+class TestDynamicShapeModels(unittest.TestCase):
+
+  def test_forward_pass_dynamic_input_correctness(self):
+    losses = []
+    for dev in [torch.device('gpu'), xla_dev]:
```

Comment:
@miladm @JackCaoG is it possible to get 2 devices on one machine? 

This tests is designed to test the model generates the same model on different devices.

Reply:
you shouldn't need this test, we expect the HLO generation part is mostly device independent.